src/password-confirmation.ts

```diff
@@ -54,13 +54,26 @@ async function _confirmPassword(password: string) {
 	logger.debug('Password confirmed')
 }
 
+// Internal state to track the dialog
+let _passwordDialog: Promise<boolean> | undefined
+let _dialogCallback: (s: string) => Promise<void>
+
 /**
  * Spawn a dialog to prompt the password.
  *
  * @param validate Is called to validate the user's password
  */
 async function promptPassword(validate: (password: string) => Promise<void>) {
-	const result = await spawnDialog(PasswordDialogVue, { validate })
+	_dialogCallback = validate
+	if (!_passwordDialog) {
+		_passwordDialog = spawnDialog(PasswordDialogVue, {
+			validate(password: string) {
+				return _dialogCallback(password)
+			},
+		})
+	}
+
+	const result = await _passwordDialog
```

Comment:
```suggestion
	const result = await _passwordDialog
	_passwordDialog = undefined
```
Reset the dialog after close?